### ISSUE TITLE
The extension context does not have a method with the name init

### ISSUE THREAD

**Issue by peteshand** [benkuper/Myo-AIR-AS3-Extension#1]
Hey there

I'm getting the following error:
[Fault] exception, information=ArgumentError: Error #3500: The extension context does not have a method with the name init.

have you seen this before?

**Comment by bilalsavas** [comment]
same issue here, were you able to fix that ?

**Comment by benkuper** [comment]
Hey,
please tell me more about your problem :
what is your OS ?
Did you get the latest version of the lib ?
If you're on OSX, did you copy myo.framework from myoSDK into /Library/Frameworks ?

**Comment by bilalsavas** [comment]
hello there, thanks for your quick response,

Im on windows 8.1, just downloaded the lib today, and using flash professional ide

I just added .ane file path to project, and I was able to get autocomplete in my as files.

seems that library tries to call .as files from a path that you specified before

ArgumentError: Error #3500: The extension context does not have a method with the name init.
    at flash.external::ExtensionContext/_call()
    at flash.external::ExtensionContext/call()
    at benkuper.nativeExtensions::MyoController()[/Users/benkuper/Documents/Dev/NativeExtensions/Myo/Myo-AS3-Extension/ane_src/as3_src/benkuper/nativeExtensions/MyoController.as:47]

Im not a pro in actionscript development environment but seeing the same issue here made me post. Maybe I just missed an easy step I should be doing, dont know.

Thanks

**Comment by benkuper** [comment]
Hi, don't worry about the path of the ActionScript file, it is how it's done when compiling libs.
What's strange is that it mentions the path of the OSX compiled lib instead of the windows one...

Please try to copy myo32.dll into your flash ide installation folder > AIR[version] > bin (just next to adl.exe) and also don't forget to include this dll in your package when you export your application.

**Comment by bilalsavas** [comment]
Oh ok, so .ane files are not cross-platform.

I added the one in the ane_src folder to my project, maybe that is OSX version one ? If so where can I find the windows one, tried to run swcCompiler but it ran with so much errors and .ane file didnt change.

btw, I couldnt find myo32.dll but theres a MyoExtension.dll, that I already put into System32 and SysWow64 folders.

**Comment by benkuper** [comment]
This is extension IS cross platform, i think it's just because last time i compiled it was from OSX, so i took the OSX path for the files, but that really doesn't matter.

You can find myo32.dll in the sdk here : https://developer.thalmic.com/login/?next=/downloads
the MyoExtension.dll does not need to be copied anywhere else, it's only used inside the ane.

**Comment by bilalsavas** [comment]
Ok, I downloaded the sdk and put myo32 & 64 dlls(just in case) in folder you mentioned. Still the same error. 

To make things clear,

I added .ane file from your "ane_src" folder to my project, (after transferring it to my project folder under /src folder)
I put myo32.dll to air16/bin folder next to adl.exe (im running air 16)

theres no other file/folder in my project folder from your library content, should there be?

or should I do anything else? should I also add swc files to project ?

**Comment by benkuper** [comment]
i just tested with flash IDE, even without adding myo32.dll and it ran just fine.
How did you add the .ane ?

**Comment by bilalsavas** [comment]
Advanced ActionScript Settings > Library Path > New > Browse to a Native Extension > Selected MyoController.ane from my projects main folder.

**Comment by bilalsavas** [comment]
I've tried in a new project with same steps but cleaner code 

package  {

```
import flash.display.MovieClip;
import benkuper.nativeExtensions.*;


public class main extends MovieClip {

    private var myoController:MyoController;        
    private var myo:Myo;        

    public function main() {
        // constructor code

        myoController = new MyoController();
        myoController.addEventListener(MyoEvent.MYO_PAIRED, myoPaired);
        myoController.setLockingPolicy(false);
    }

    private function myoPaired(e:MyoEvent):void 
    {
        trace("Myo is paired");
        myo = e.myo;
        myo.setLock(false);

    }
}
```

}

same error happens, are you using same .ane with the one uploaded here?

**Comment by benkuper** [comment]
can you try with FlashDevelop ?

**Comment by bilalsavas** [comment]
never used it before but i'll try and let you know, 

thanks again

**Comment by bilalsavas** [comment]
Im still in flash ide, I renamed .ane to .swc and added that way, this time im getting null object reference errors (ofcourse) but theres also this : "[MyoController] Problem with creating ANE" in trace log, does that mean anything, can this be a clue ?

**Comment by bilalsavas** [comment]
Hi ben, 

After many trials I found a weird solution, I dont know how its linked with the issue but its solved, I'm able to build and run in both FlashDevelop & Flash PRO IDE' s

I started trying stuff with FlashDevelop since my last post, installed a few apps, did some configurations in windows (for command line functions to work), Installed flex + air in flash develop, did the path setting in bat files, then I was able to run the app but it gave me "pthreadVC2.dll error is missing" error.

It seemed irrelevant at first, but having no other way to solve made me try, so, I found that dll on apache/php site, and put that in to air[version]/bin folder you mentioned before and it worked.

link to dll, (if anyone else came across this issue, they might solve it that way) ;
http://windows.php.net/downloads/pecl/releases/pthreads/2.0.9/
The one I used is "php_pthreads-2.0.9-5.6-ts-vc11-x86.zip"

(Might be an OS version issue mine is Win 8.1)

Thank you for the library!
